lib/api/CFieldDataCategorizer.cc

```diff
@@ -30,30 +30,30 @@ namespace api {
 
 namespace {
 
-const std::string VERSION_TAG("a");
-const std::string CATEGORIZER_TAG("b");
-const std::string EXAMPLES_COLLECTOR_TAG("c");
+const std::string VERSION_TAG{"a"};
+const std::string CATEGORIZER_TAG{"b"};
+const std::string EXAMPLES_COLLECTOR_TAG{"c"};
 } // unnamed
 
 // Initialise statics
-const std::string CFieldDataCategorizer::ML_STATE_INDEX(".ml-state");
-const std::string CFieldDataCategorizer::MLCATEGORY_NAME("mlcategory");
-const double CFieldDataCategorizer::SIMILARITY_THRESHOLD(0.7);
-const std::string CFieldDataCategorizer::STATE_TYPE("categorizer_state");
-const std::string CFieldDataCategorizer::STATE_VERSION("1");
+const std::string CFieldDataCategorizer::ML_STATE_INDEX{".ml-state"};
+const std::string CFieldDataCategorizer::MLCATEGORY_NAME{"mlcategory"};
+const double CFieldDataCategorizer::SIMILARITY_THRESHOLD{0.7};
+const std::string CFieldDataCategorizer::STATE_TYPE{"categorizer_state"};
+const std::string CFieldDataCategorizer::STATE_VERSION{"1"};
 
 CFieldDataCategorizer::CFieldDataCategorizer(const std::string& jobId,
                                              const CFieldConfig& config,
                                              model::CLimits& limits,
                                              COutputHandler& outputHandler,
                                              CJsonOutputWriter& jsonOutputWriter,
                                              CPersistenceManager* persistenceManager)
-    : m_JobId(jobId), m_Limits(limits), m_OutputHandler(outputHandler),
-      m_ExtraFieldNames(1, MLCATEGORY_NAME), m_WriteFieldNames(true),
-      m_NumRecordsHandled(0), m_OutputFieldCategory(m_Overrides[MLCATEGORY_NAME]),
-      m_MaxMatchingLength(0), m_JsonOutputWriter(jsonOutputWriter),
-      m_CategorizationFieldName(config.categorizationFieldName()),
-      m_CategorizationFilter(), m_PersistenceManager(persistenceManager) {
+    : m_JobId{jobId}, m_Limits{limits}, m_OutputHandler{outputHandler},
+      m_ExtraFieldNames{1, MLCATEGORY_NAME}, m_WriteFieldNames{true},
+      m_NumRecordsHandled{0}, m_OutputFieldCategory{m_Overrides[MLCATEGORY_NAME]},
+      m_MaxMatchingLength{0}, m_JsonOutputWriter{jsonOutputWriter},
+      m_CategorizationFieldName{config.categorizationFieldName()},
+      m_CategorizationFilter{}, m_PersistenceManager{persistenceManager} {
```

Comment:
nit: there doesn't seem to be any need to default initialise `m_CategorizationFilter` here. Also, how about initialising builtin types `m_NumRecordsHandled`, etc in the class body?